dotCMS/src/main/java/com/dotmarketing/factories/PublishFactory.java

```diff
@@ -103,6 +103,29 @@ public static boolean publishAsset(Inode webAsset,HttpServletRequest req) throws
 		return publishAsset(webAsset, user, false);
 	}
 
+
+	/**
+	 * This method publish a given folder and its related assets, if a user is passed the method will check permissions to publish,
+	 * if the user doesn't have permission to publish one of the related assets then that one will be skipped.
+	 *
+	 * @param folder
+	 * @param req
+	 * @return
+	 * @throws WebAssetException
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 */
+	public static boolean publishAsset(Folder folder,HttpServletRequest req) throws WebAssetException, DotSecurityException, DotDataException {
```

Comment:
Not sure if the request should be here, is there a way to pass the user and retrieve it from upper layer that have natural access to the request.

Reply:
I just refactored the class. The logic had always been there

Reply:
I think it's a good idea to implement your changes in a separate PR